### PR TITLE
Check the first dimension each time for any layout

### PR DESCRIPTION
**Pull request by pereanub**
### Details:
 - *Check the first dimension each time for any layout*
 - *Get batch size only when inputs/outputs of the compiled model are equal to 1*

### Tickets:
 - *E#103116*

### CODE REVIEW

File: src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp [hunk 1]
```diff
@@ -59,26 +59,22 @@ std::optional<size_t> getBatchSizeForNode(const IONodeDescriptor& nodeDescriptor
                                           const ZeroExecutor::ArgumentDescriptor& zeDescriptor) {
     Logger logger("GetBatchSizeForNode", Logger::global().level());
 
+    if (nodeDescriptor.originalShape.is_dynamic() || nodeDescriptor.originalShape.rank().get_length() == 0) {
+        logger.info("Dynamic networks are not supported when batching is handled by the plugin");
```
Comment (razvanapetroaie) [inline src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp:63]: I find the logger message a bit confusing in the `nodeDescriptor.originalShape.rank().get_length() == 0` case, maybe we should display different messages for different corner cases.

File: src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp [hunk 2]
```diff
@@ -59,26 +59,22 @@ std::optional<size_t> getBatchSizeForNode(const IONodeDescriptor& nodeDescriptor
...
-        logger.info("Batching on the plugin is working only when batching is found on 0th dimension");
+
+    if ((ovDimensions[BATCH_AXIS] == zeDescriptor.info.dims[BATCH_AXIS]) &&
+        (ovDimensions[BATCH_AXIS] != DEFAULT_BATCH_SIZE)) {
+        logger.info("Batching on the plugin is not used, batching is handled by the compiler");
         return std::nullopt;
     }
 
+    if (zeDescriptor.info.dims[BATCH_AXIS] == DEFAULT_BATCH_SIZE)
+        return ovDimensions[BATCH_AXIS];
```
[the remark this answers is not in the document]
Comment (pereanub) [reply src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp]: done, thanks

File: src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp (hunk 1 above)
Comment (pereanub) [reply src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp:68]: created a separate check and message for this case. thanks!

**Review by razvanapetroaie — APPROVED** [review]

File: src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp [hunk 3]
```diff
@@ -59,26 +59,28 @@ std::optional<size_t> getBatchSizeForNode(const IONodeDescriptor& nodeDescriptor
...
-            return std::nullopt;
-        } else {
-            return ovDimensions[BATCH_AXIS];
-        }
-        break;
-    default:
-        logger.info("Batching on the plugin is working only when batching is found on 0th dimension");
+
+    if ((ovDimensions[BATCH_AXIS] == zeDescriptor.info.dims[BATCH_AXIS]) &&
+        (ovDimensions[BATCH_AXIS] != DEFAULT_BATCH_SIZE)) {
```
Comment (ArtemySkrebkov) [inline src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp:75]: Nitpick: too much brackets for my taste
```suggestion
    if (ovDimensions[BATCH_AXIS] == zeDescriptor.info.dims[BATCH_AXIS] &&
        ovDimensions[BATCH_AXIS] != DEFAULT_BATCH_SIZE) {
```

**Review by ArtemySkrebkov — COMMENTED** [review]
Is there a way to test the change?

**Review by ArtemySkrebkov — APPROVED** [review]

**Review by PatrikStepan — APPROVED** [review]

**Pull request merged**